model_versions.py

```diff
@@ -39,18 +29,20 @@ def gen_spack_pack_list():
     uv = jetlag_conf.get_uv()
     os.system("rm -fr input.tgz run_dir")
     os.system("mkdir -p run_dir")
-    with HiddenPrint():
+    with open("spack-log.txt", "w") as f:
+        default_stdout = sys.stdout
+        sys.stdout = f
```

Comment:
Log files should be opened for append, i.e. open("file", "a"). Also, using a class like HiddenPrint to restore sys.stdout is good practice and safer. Otherwise, an exception in the intervening code can lead to sys.stdout being lost.